File: TemaProiect/motoProjectC#/motoProjectCSharp/motoProjectCSharp/domain/Team.cs
namespace motoProjectCSharp.domain;

using System.Collections.Generic;

public class Team : Entity<long>
{
    public string Name { get; set; }
    public List<Participant> TeamMembers { get; set; }

    public Team(long id, string name, List<Participant> teamMembers) : base(id)
    {
        Name = name;
        TeamMembers = teamMembers;
    }
}

File: TemaProiect/motoProject/src/main/java/org/example/motoproject/domain/Team.java
package org.example.motoproject.domain;

import java.util.ArrayList;
import java.util.List;

public class Team extends Entity<Long> {
    private String name;
    private List<Participant> teamMembers = new ArrayList<>();

    public Team(Long id, String name, List<Participant> teamMembers) {
        super(id);
        this.name = name;
        this.teamMembers = teamMembers;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Participant> getTeamMembers() {
        return teamMembers;
    }

    public void setTeamMembers(List<Participant> teamMembers) {
        this.teamMembers = teamMembers;
    }
}
